### PR TITLE
Update PR template

### PR DESCRIPTION
**Pull request by simitt**
## Motivation/summary
After working with the PR template for a while, revisit what is most useful for a checklist and what should be moved to a comment section or removed all together.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -21,40 +21,41 @@ See also https://github.com/elastic/apm-server/blob/master/CONTRIBUTING.md for m
...
-<!--
+
+Ensure your code follows the style guidelines of this project (run `make check-full` for static code checks and linting), and rebase your changes on top of the latest master branch.
+
 Run the test suite to make sure that nothing is broken. See https://github.com/elastic/apm-server/blob/master/TESTING.md for details.
 -->
+
+I have ensured my changes include/I have considered implications for:
+- [ ] documentation
+- [ ] logging (add log lines, chose appropriate log selector, etc.)
```
Comment (axw) [inline .github/pull_request_template.md:40]:
```suggestion
- [ ] logging (add log lines, choose appropriate log selector, etc.)
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -21,40 +21,41 @@ See also https://github.com/elastic/apm-server/blob/master/CONTRIBUTING.md for m
...
+- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations)
```
Comment (axw) [inline .github/pull_request_template.md:41]: Would be good to reference a doc or example PR, if or when we have one.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -21,40 +21,41 @@ See also https://github.com/elastic/apm-server/blob/master/CONTRIBUTING.md for m
...
+- [ ] automated tests (add tests for the code changes, all [**unit** tests](https://github.com/elastic/apm-server/blob/master/TESTING.md) pass locally)
+- [ ] telemetry
```
Comment (axw) [inline .github/pull_request_template.md:43]: same for telemetry

**Comment by simitt** [comment]
@jalvz would be great to also get your opinion on these changes.

**Comment by simitt** [comment]
Thanks for the suggestions! I removed the comments from the single sections for less distraction when filling out the PR. 
But I did move them into one big comment at the beginning of the template. IMO these comments are not only helpful for first time contributors, e.g. we also sometimes forget to add labels, run the `make check-full` etc. Having the suggestions summarized directly in the PR template makes them more obvious than having them in an extra file. For the sake of readability, I did shorten the comments a lot though, and hope they are easier to digest now. 

Let me know what you think.

**Pull request merged**